Muon DIS with Pythia6 run through Geant4

siilieva opened the pull request:
Create a FairTask that injects a custom process into Geant4 (since we run the MC engine through a FairRunSim instance)
In this case the process is muonDIS  and is being generated using Pythia6.

The muonDIS procedure is technically a C++ copy of the muon dis generation code from  [muonDIS.py](https://github.com/SND-LHC/sndsw/blob/c0fe18fad8300fa46a7bfee53733f20e0652994d/shipLHC/muonDis.py#L448-L528)

Also, the famous [MeanMaterianBudget](https://github.com/SND-LHC/sndsw/blob/c0fe18fad8300fa46a7bfee53733f20e0652994d/shipgen/MuDISGenerator.cxx#L56-L204) is made public and is widely used in the code. A tiny update was made there allowing the geo navigator to step forward shall a particle be moving along a volume border.

Tuning was done to make sure the proposed simulation chain works correctly and efficiently! See related presentations: [2July2025](https://indico.cern.ch/event/1563064/contributions/6590880/attachments/3096944/5486599/muonDIS_MC_2July_2025.pdf) and [30July2025](https://indico.cern.ch/event/1563072/contributions/6633823/attachments/3112593/5517800/muonDIS_MC_30July_2025.pdf) and [10September2025](https://indico.cern.ch/event/1563075/contributions/6690909/attachments/3132941/5558716/muonDIS_MC_10September_2025.pdf).

On top of the previously available option to set z range for the DIS vertex, the new chain allows to fully simulate events only if DIS vertex is in a specific x-y range AND/OR a volume. This saves a lot of computing resources.

Revision of indices of TPythia6 arrays  was done to correct a hiccup with the cross section, but that doesn't affect previous MC chain, just made obtaining this more difficult.

Creating draft to show result on SoftPhys meeting on 27.08.2025 first, then will go public.

ps. candidate for  cherry-picking in SHiP
ps. an internal/technical note is to be prepared

Commit:
Add MuonDISProcessInjector task

It adds to the Geant4 process manager a custom process
that is using Pythia6 to generate muonDIS.

Commit:
Make the MeanMaterialBudget calculation public

to reuse the function globally in sndsw.

Commit:
run_simSND: Add options to run muonDIS with Geant4

Commit:
Add muon DIS generated my Pythia6 as a Geant4 process

Using standalone CMake and don't rely on the FairRoot CMake macros
as we usually do.

Commit:
run_simSND: update logger settings

Better control of severity level and make the log print in colour.

Commit:
MeanMaterialBudget: Unblock navigation along volume borders

In rare cases, a track is moving right along the border of two volumes.
This makes the geo navigation stuck. To unblock it, one now moves along
the particle trajectory with a step of 1e-6cm, smaller than any position
measurement precision in SND. This way, one allows the navigator to move
away from any border at a reasonable computing cost.

Commit:
Fix the direct extraction of pythia xsec from the code

When going from original p6 code(Fortran) to ROOT's TPyhia6 some arrays, like the XSEC one,
have their indices swapped, also counting starts from 0 in TPythia6 case. With this fix one can
read the P6 xsec directly from the code, no need to look into the produced txt file.
Another part of the fix is to allow using P6 or P8 when creating the DIS xsec file using existing
CL options.

Commit:
Allow setting Ekin cut when using FastMuon option

This is essential when running muonDIS with activated emulsions.

olantwin reviewed (commented):
A first batch of comments. Impressive work!

Commit:
Add macro to calculate Pythia6 xsec over N events

Using the pre-existing option in muonDis.py to do that task, the process often
hangs shall there be many events, order of 1M. A dedicated macro is now added,
mimicking the generator options used before, but where one runs the event
generation as a separate process to monitor its progress and recover shall
it get stuck.
